src/game/shell.c

```diff
@@ -300,16 +301,37 @@ void Shell_ProcessInput(void)
     if (g_InputDB.toggle_bilinear_filter) {
         g_Config.rendering.texture_filter =
             (g_Config.rendering.texture_filter + 1) % GFX_TF_NUMBER_OF;
+
+        switch (g_Config.rendering.texture_filter) {
+        case GFX_TF_NN:
+            Console_Log(GS(OSD_TEXTURE_FILTER_SET), GS(OSD_TEXTURE_FILTER_NN));
+            break;
+        case GFX_TF_BILINEAR:
+            Console_Log(
+                GS(OSD_TEXTURE_FILTER_SET), GS(OSD_TEXTURE_FILTER_BILINEAR));
+            break;
+        case GFX_TF_NUMBER_OF:
+            break;
+        }
+
         Config_Write();
     }
 
     if (g_InputDB.toggle_perspective_filter) {
         g_Config.rendering.enable_perspective_filter ^= true;
+        Console_Log(
+            g_Config.rendering.enable_perspective_filter
+                ? GS(OSD_PERSPECTIVE_FILTER_ON)
+                : GS(OSD_PERSPECTIVE_FILTER_OFF));
         Config_Write();
     }
 
     if (g_InputDB.toggle_fps_counter) {
         g_Config.rendering.enable_fps_counter ^= true;
+        Console_Log(
+            g_Config.rendering.enable_perspective_filter
```

Comment:
This is the only bug I noticed. The FPS message is incorrectly linked to the `enable_perspective_filter` option.